### ISSUE TITLE
PT2E Dynamic Quantization with XNNPACK Backend for Android fails Loading of method forward in runtime

### ISSUE THREAD

**Issue by 24hari1998** [pytorch/executorch#11355]
### 🐛 Describe the bug

I am trying to perform a Post Training Dynamic Quantization with PyTorch 2 Export method for LayoutLMV3 (multimodal transformer with 1d and 2d bias in attention) open source model. The conversion code is as below:

[Convertion and Lowering code]
'''"
dynamic_qconfig = get_symmetric_quantization_config(is_dynamic=True, is_per_channel=True)
quantizer = XNNPACKQuantizer()
quantizer.set_global(dynamic_qconfig)
exported_program = export_for_training(model, inputs_tuple).module()
output_exported = exported_program(example_inputs)
model_prepared = prepare_pt2e(exported_program, quantizer)
quantized_model = convert_pt2e(model_prepared)
exported_quantized = export(quantized_model, inputs_tuple)
 edge: EdgeProgramManager = to_edge_transform_and_lower(exported_quantized, partitioner = [XnnpackPartitioner()])
et = edge.to_executorch()
"""
Run command:
"./cmake-out/backends/xnnpack/xnn_executor_runner --model_path=./model_quant_debug.pte"

Error:
I 00:00:00.002518 executorch:cpuinfo_utils.cpp:91] Reading file /sys/devices/system/cpu/cpu0/regs/identification/midr_el1
I 00:00:00.002529 executorch:cpuinfo_utils.cpp:100] Failed to open midr file /sys/devices/system/cpu/cpu0/regs/identification/midr_el1
I 00:00:00.002535 executorch:cpuinfo_utils.cpp:116] CPU info and manual query on # of cpus dont match.
I 00:00:00.002545 executorch:executor_runner.cpp:142] Resetting threadpool with num threads = 0
I 00:00:00.006525 executorch:executor_runner.cpp:166] Model file ./model_quant_debug.pte is loaded.
I 00:00:00.006541 executorch:executor_runner.cpp:175] Using method forward
I 00:00:00.006564 executorch:executor_runner.cpp:226] Setting up planned buffer 0, size 311091824.
Error in XNNPACK: failed to create Channelwise Quantized Dense Tensor value with 1.653039e-39 scale in channel #1: scale must be finite, normalized, and positive (xnn_validate_channelwise_quantized_tensor, /storage/Harinarayanan_K/executorch/executorch/backends/xnnpack/third-party/XNNPACK/src/tensor.c:385)
E 00:00:00.255082 executorch:XNNCompiler.cpp:538] Failed to define tensor 2 with code: xnn_status_invalid_parameter
E 00:00:00.255121 executorch:XNNPACKBackend.cpp:119] XNNCompiler::compileModel failed: 0x1
E 00:00:00.255129 executorch:method.cpp:109] Init failed for backend XnnpackBackend: 0x1
F 00:00:00.255158 executorch:executor_runner.cpp:246] In function main(), assert failed (method.ok()): Loading of method forward failed with status 0x1

What could possible cause this error ? Is it because of any unsupported operations or any error in the conversion steps. Kindly requesting advice on the error above, thanks.

### Versions

PyTorch version: 2.6.0+cu124
Is debug build: False
CUDA used to build PyTorch: 12.4
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.5 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.30.1
Libc version: glibc-2.35

Python version: 3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-6.8.0-45-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 12.6.85
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: NVIDIA RTX A6000
GPU 1: NVIDIA RTX A6000
GPU 2: NVIDIA RTX A6000
GPU 3: NVIDIA RTX A6000

Nvidia driver version: 560.35.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        52 bits physical, 57 bits virtual
Byte Order:                           Little Endian
CPU(s):                               112
On-line CPU(s) list:                  0-111
Vendor ID:                            GenuineIntel
Model name:                           Intel(R) Xeon(R) w9-3495X
CPU family:                           6
Model:                                143
Thread(s) per core:                   2
Core(s) per socket:                   56
Socket(s):                            1
Stepping:                             8
CPU max MHz:                          4800.0000
CPU min MHz:                          800.0000
BogoMIPS:                             3792.00
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 intel_ppin cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect user_shstk avx_vnni avx512_bf16 wbnoinvd dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req vnmi avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm md_clear serialize tsxldtrk pconfig arch_lbr ibt amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities
Virtualization:                       VT-x
L1d cache:                            2.6 MiB (56 instances)
L1i cache:                            1.8 MiB (56 instances)
L2 cache:                             112 MiB (56 instances)
L3 cache:                             105 MiB (1 instance)
NUMA node(s):                         1
NUMA node0 CPU(s):                    0-111
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] nvidia-cublas-cu11==11.10.3.66
[pip3] nvidia-cublas-cu12==12.4.5.8
[pip3] nvidia-cuda-cupti-cu11==11.7.101
[pip3] nvidia-cuda-cupti-cu12==12.4.127
[pip3] nvidia-cuda-nvrtc-cu11==11.7.99
[pip3] nvidia-cuda-nvrtc-cu12==12.4.127
[pip3] nvidia-cuda-runtime-cu11==11.7.99
[pip3] nvidia-cuda-runtime-cu12==12.4.127
[pip3] nvidia-cudnn-cu11==8.5.0.96
[pip3] nvidia-cudnn-cu12==9.1.0.70
[pip3] nvidia-cufft-cu11==10.9.0.58
[pip3] nvidia-cufft-cu12==11.2.1.3
[pip3] nvidia-curand-cu11==10.2.10.91
[pip3] nvidia-curand-cu12==10.3.5.147
[pip3] nvidia-cusolver-cu11==11.4.0.1
[pip3] nvidia-cusolver-cu12==11.6.1.9
[pip3] nvidia-cusparse-cu11==11.7.4.91
[pip3] nvidia-cusparse-cu12==12.3.1.170
[pip3] nvidia-cusparselt-cu12==0.6.2
[pip3] nvidia-nccl-cu11==2.14.3
[pip3] nvidia-nccl-cu12==2.21.5
[pip3] nvidia-nvjitlink-cu12==12.4.127
[pip3] nvidia-nvtx-cu11==11.7.91
[pip3] nvidia-nvtx-cu12==12.4.127
[pip3] onnx==1.14.0
[pip3] onnxruntime==1.15.1
[pip3] open_clip_torch==2.26.1
[pip3] optree==0.12.1
[pip3] pytorch-metric-learning==2.6.0
[pip3] torch==2.6.0
[pip3] torchaudio==2.0.2
[pip3] torchvision==0.15.2
[pip3] triton==3.2.0
[conda] Could not collect

**Comment by 24hari1998** [comment]
cc @mergennachin  @iseeyuan  @lucylq  @helunwencser  @tarun292  @kimishpatel  @jackzhxng  @kirklandsign  @cbilgin @larryliu0820  @cccclai

**Comment by mcr229** [comment]
hi can you try calibrating the model before exporting? you should be able to do this by running a couple sample inputs through the prepared graph. Like such:
```

for i in range(0, 10);
    prepared_model(*sample_inputs)
```

**Comment by 24hari1998** [comment]
Thank for the tip @mcr229 , its working after trying calibration before exporting. Closing the issue.
